03week/towersOfHanoi.js

```diff
@@ -13,42 +13,91 @@ let stacks = {
   c: []
 };
 
+/*
+*This function displays the "board" to the user
+*/
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+/*
+*This function removes a block from a stack and adds that block to a different stack
+*
+* @param startStack the stack we are removing the element from
+* @param endStack the stack we are adding the element to
+*/
+function movePiece(startStack, endStack) {
+  let blockMove = stacks[startStack].pop(); //remove the "top block" from startStack
+  stacks[endStack].push(blockMove); //add the removed block to endStack
 }
 
-function isLegal() {
-  // Your code here
-
+/*
+*This function checks to see if the proposed move is legal
+*
+* @param startStack the stack we are removing the element from
+* @param endStack the stack we are adding the element to
+* @return true if the move is legal
+*/
+function isLegal(startStack, endStack) {
+  if (((startStack==='a')||(startStack==='b')||(startStack==='c'))&&
+     ((endStack==='a')||(endStack==='b')||(endStack==='c'))) { //checking to make sure input is valid
+    let begin = stacks[startStack];
+    let final = stacks[endStack];
+    //console.log("Length of startStack: " + begin.length);
+    //console.log("Value of block being moved: " + begin[begin.length - 1]);
+    //console.log("Value of block being topped off: " + final[final.length - 1]);
+    if (begin[begin.length - 1] > final[final.length - 1]) { //making sure the "block" is being move onto a larger "block"
```

Comment:
this should return a false if it is not a legal move.
Also checking for a legal move is a better approach.  IE start out assuming the move is not legal unless you prove otherwise.

This implementation allows you to move piece back to its own stack. Not wrong just weird that it would be allowed.